test/widgets/lightbox_test.dart

```diff
@@ -1,77 +1,148 @@
 import 'dart:async';
 
 import 'package:checks/checks.dart';
+import 'package:clock/clock.dart';
+import 'package:flutter/services.dart';
 import 'package:flutter_gen/gen_l10n/zulip_localizations.dart';
 import 'package:flutter_test/flutter_test.dart';
 import 'package:flutter/material.dart';
 import 'package:plugin_platform_interface/plugin_platform_interface.dart';
 import 'package:video_player_platform_interface/video_player_platform_interface.dart';
 import 'package:video_player/video_player.dart';
+import 'package:zulip/model/localizations.dart';
 import 'package:zulip/widgets/lightbox.dart';
 import 'package:zulip/widgets/store.dart';
 
 import '../example_data.dart' as eg;
 import '../model/binding.dart';
+import 'dialog_checks.dart';
+
+const kTestVideoUrl = "https://a/video.mp4";
+const kTestUnsupportedVideoUrl = "https://a/unsupported.mp4";
+const kTestVideoDuration = Duration(seconds: 10);
 
 class FakeVideoPlayerPlatform extends Fake
     with MockPlatformInterfaceMixin
     implements VideoPlayerPlatform {
-  static const int _textureId = 0xffffffff;
-
-  static StreamController<VideoEvent> _streamController = StreamController<VideoEvent>();
-  static bool initialized = false;
-  static bool isPlaying = false;
+  static final FakeVideoPlayerPlatform instance = FakeVideoPlayerPlatform();
 
   static void registerWith() {
-    VideoPlayerPlatform.instance = FakeVideoPlayerPlatform();
+    VideoPlayerPlatform.instance = instance;
+  }
+
+  static const int _kTextureId = 0xffffffff;
+
+  StreamController<VideoEvent> _streamController = StreamController<VideoEvent>();
+  bool _hasError = false;
+  Duration _lastSetPosition = Duration.zero;
+  Stopwatch? _stopwatch;
+
+  List<String> get callLog => _callLog;
+  final List<String> _callLog = [];
+
+  bool get initialized => _initialized;
+  bool _initialized = false;
+
+  bool get isCompleted => _isCompleted;
+  bool _isCompleted = false;
+
+  bool get isPlaying => _stopwatch?.isRunning ?? false;
+
+  Duration get position {
+    assert(_stopwatch != null);
+    final pos = _stopwatch!.elapsed + _lastSetPosition;
+    if (pos.compareTo(kTestVideoDuration) >= 0) {
+      return kTestVideoDuration;
+    } else {
+      return pos;
+    }
```

Comment:
```suggestion
    return pos <= kTestVideoDuration ? pos : kTestVideoDuration;
```

Or is there a subtle difference between that and `compareTo` which I'm missing?